aether-entity-extraction-module/extractor/utils.py

```diff
@@ -65,22 +69,30 @@ class Task(NamedTuple):
     data: Union[Dict, None] = None
 
 
-def kernel_data_request(url='', method='get', data=None, headers=None):
+def kernel_data_request(url='', method='get', data=None, headers=None, realm=None):
     '''
     Handle request calls to the kernel server
     '''
+
+    kernel_url = settings.KERNEL_URL
+    headers = headers or {}
+    headers['Authorization'] = f'Token {settings.KERNEL_TOKEN}'
+    _realm = realm if realm else settings.GATEWAY_PUBLIC_REALM
+
+    headers[settings.REALM_COOKIE] = _realm
+
     res = request(
         method=method,
-        url=f'{settings.KERNEL_URL}/{url}',
+        url=f'{kernel_url}/{url}',
         json=data or {},
-        headers={'Authorization': f'Token {settings.KERNEL_TOKEN}'},
+        headers=headers,
     )
 
     res.raise_for_status()
     return json.loads(res.content.decode('utf-8'))
 
 
-def get_from_redis_or_kernel(id, type, tenant, redis=None):
+def get_from_redis_or_kernel(id, _type, tenant, redis=None):
```

Comment:
maybe `model`? or `model_type`? There is a convention in python that the names starting with underscore are private :woman_shrugging: